coderd/httpmw/apikey.go

```diff
@@ -240,6 +240,16 @@ func ExtractAPIKey(rw http.ResponseWriter, r *http.Request, cfg ExtractAPIKeyCon
 		})
 	}
 
+	// Validate OAuth2 provider app token audience (RFC 8707) if applicable
+	if key.LoginType == database.LoginTypeOAuth2ProviderApp {
+		if err := validateOAuth2ProviderAppTokenAudience(ctx, cfg.DB, *key, r); err != nil {
+			return optionalWrite(http.StatusForbidden, codersdk.Response{
+				Message: "Token audience mismatch",
+				Detail:  err.Error(),
```

Comment:
Does this potentially leak information? Should we maybe log this instead but not return it?